src/tools/miri/src/helpers.rs

```diff
@@ -32,6 +32,8 @@ pub enum AccessKind {
 ///
 /// A `None` namespace indicates we are looking for a module.
 fn try_resolve_did(tcx: TyCtxt<'_>, path: &[&str], namespace: Option<Namespace>) -> Option<DefId> {
+    let _span = enter_trace_span!(resolve::try_resolve_did, ?path);
```

Comment:
This isn't the same kind of operation as the others at all, it is using `resolve` to refer to something else. It's a good idea to trace this but separate from the normal `resolve` calls.

Reply:
Ok, I will call it just "try_resolve_did" then. I thought this belonged to the same category because it seemed to resolve instances from module paths.

Reply:
Yeah but the the resolves resolve instances from DefIds (by doing trait solving)... whereas this here *produces* a DefId (by doing basically name resolution). It's entirely different machinery.